src/datahike/query.cljc

```diff
@@ -87,6 +88,15 @@
 
 ;; Utilities
 
+(defn distinct-tuples [tuples]
```

Comment:
Good point. The reason for this function is that it seemed that tuples are sometimes represented by Java object arrays (supposedly for performance reasons). And object arrays are equal only if their JVM references are equal. Because this function compares them by the values of applying `vec`, two different object arrays will be considered equal if they contain the same elements. I wrote an explanation in the code and improved the unit test.